### PR TITLE
lefthook 0.7.3 (new formula)

### PR DESCRIPTION
**Pull request by skryukov**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by nandahkrishna — COMMENTED** [review]
Thanks for your PR, @skryukov. Please see my comment below.

File: Formula/lefthook.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    mkdir_p testpath/".git/hooks"
+    system bin/"lefthook install"
+
+    assert_predicate testpath/"lefthook.yml", :exist?
+    assert_match "#{version}", shell_output("#{bin}/lefthook version").strip
```
Comment (SMillerDev) [inline Formula/lefthook.rb:19]: This shouldn't be needed. `assert_match` also allows partial matches
```suggestion
    assert_match "#{version}", shell_output("#{bin}/lefthook version")
```

File: Formula/lefthook.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  desc "Fast and powerful Git hooks manager for any type of projects"
+  homepage "https://github.com/evilmartians/lefthook"
+  url "https://github.com/evilmartians/lefthook/archive/refs/tags/v0.7.3.tar.gz"
+  sha256 "643bb3d2a1b67d865285a9cadcad11f2588ba284570ab9924c8e3d2e902301ca"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
```
Comment (chenrui333) [inline Formula/lefthook.rb:11]:
```suggestion
    system "go", "build", "-ldflags", "-s -w", *std_go_args
```

File: Formula/lefthook.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  test do
+    system "git", "init"
+    system bin/"lefthook install"
```
Comment (chenrui333) [inline Formula/lefthook.rb:16]:
```suggestion
    system bin/"lefthook", "install"
```

**Review by chenrui333 — COMMENTED** [review]
Some minor suggestions. :)

File: Formula/lefthook.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  desc "Fast and powerful Git hooks manager for any type of projects"
+  homepage "https://github.com/evilmartians/lefthook"
+  url "https://github.com/evilmartians/lefthook/archive/refs/tags/v0.7.3.tar.gz"
+  sha256 "643bb3d2a1b67d865285a9cadcad11f2588ba284570ab9924c8e3d2e902301ca"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags", "-s -w", *std_go_args
```
Comment (carlocab) [inline Formula/lefthook.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w")
```
Let's use the newer syntax here. CC @chenrui333

**Review by carlocab — APPROVED** [review]
Thanks, @skryukov.

**Pull request closed without merging**